daemon/src/org/eclipse/iofog/resource_consumption_manager/ResourceConsumptionManager.java

```diff
@@ -166,6 +174,38 @@ private float getCpuUsage() {
 		}
 	}
 
+	private long getSystemAvailableMemory() {
+	    if (SystemUtils.IS_OS_WINDOWS) {
+	        return 0;
+        }
+		final String MEM_AVAILABLE = "grep 'MemAvailable' /proc/meminfo | awk '{print $2}'";
+		CommandShellResultSet<List<String>, List<String>> resultSet = executeCommand(MEM_AVAILABLE);
+		long memInKB = Long.parseLong(parseOneLineResult(resultSet));
+		return memInKB * 1024;
+	}
+
+	private float getTotalCpu() {
+        if (SystemUtils.IS_OS_WINDOWS) {
+            return 0;
+        }
+		final String CPU_USAGE = "grep 'cpu' /proc/stat | awk '{usage=($2+$3+$4)*100/($2+$3+$4+$5+$6+$7+$8+$9)} END {print usage}'";
```

Comment:
add here comment with link: https://github.com/Leo-G/DevopsWiki/wiki/How-Linux-CPU-Usage-Time-and-Percentage-is-calculated